ext/openssl/ossl_ssl.c

```diff
@@ -2065,14 +2065,16 @@ ossl_ssl_write_internal(VALUE self, VALUE str, VALUE opts)
     if (!ssl_started(ssl))
         rb_raise(eSSLError, "SSL session is not started yet");
 
-    tmp = rb_str_new_frozen(StringValue(str));
+    tmp = rb_str_locktmp(StringValue(str));
```

Comment:
I think we want to skip `rb_str_locktmp()` if the String is already frozen. It currently works, but locking a frozen String seems strange. I think we're safe to assume `RSTRING_PTR(str)` won't change.

Reply:
We could indeed, but that makes the codepath more complicated for not much gain? Except maybe avoid mutating a frozen string whihc could have some small copy on write consequences. Perhaps it's more of a job for `rb_str_locktmp` to skip the lock if the string is already frozen?

Reply:
I think rb_str_locktmp() is not well defined for frozen strings (e.g. https://github.com/oracle/truffleruby/issues/3752) so it seems best to avoid it.